### ISSUE TITLE
[BUG]

### ISSUE THREAD

**Issue by Rabave3** [microsoft/playwright#26351]
Having two test files i.e., test1.spec.js and test2.spec.js. If I run both the file at a time in the index.html file I can see only test2.spec.js results. test1.spec.js result is stored in the index.html file but if the test2.spec.js file is run then the result is overwritten in the same index.html file. I need both the test results in the index.html file.

**Comment by dgozman** [comment]
We have to reproduce this issue locally to be able to debug it. If you can provide a full repro, please file a new issue by filling the "Bug Report" template, and link to this one.

**Comment by Rabave3** [comment]
Package.json

{
  "name": "perxcept",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "Extension": "node extensionScript.js",
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "@playwright/test": "^1.36.2",
    "eslint": "^8.28.0",
    "eslint-config-google": "^0.14.0"
  },
  "dependencies": {
    "csv-parse": "^5.3.2",
    "dotenv": "^16.0.3",
    "fs": "^0.0.1-security",
    "knex": "^2.4.2",
    "mysql": "^2.18.1",
    "mysql2": "^3.1.2",
    "properties-reader": "^2.2.0",
    "xlsx": "^0.18.5"
  }
}

extensionScript.js

const {spawnSync} = require('child_process');

const testFiles = [
    'emailForwarding.spec.js',
    'webcontent.spec.js',
    // Add more test files as needed in the desired execution order
];

// Helper function to run a single test file
function runTest(testFile) {
    const result = spawnSync('npx',
        ['playwright', 'test', '--config', 'extension.config.js', testFile], {
            stdio: 'inherit',
        });

    if (result.error || result.status !== 0) {
        console.error(`Test file "${testFile}" failed.`);
    }
}

// Execute tests sequentially
for (const testFile of testFiles) {
    runTest(testFile);
}

// Exit the process once all test files are executed
process.exit(0);

extension.config.js

// @ts-check
const {devices} = require('@playwright/test');
// import BrowserContext, { chromium } from "@playwright/test";
/**
 * Read environment variables from file.
 * https://github.com/motdotla/dotenv
 */
require('dotenv').config();

// const browserTypes = {chromium}
/**
 * @see https://playwright.dev/docs/test-configuration
 * @type {import('@playwright/test').PlaywrightTestConfig}
 */

const config = {
    testDir: './tests/Perxcept/Extension',
    /* Maximum time one test can run for. */
    timeout: 90 * 1000,
    expect: {
    /**
     * Maximum time expect() should wait for the condition to be met.
     * For example in `await expect(locator).toHaveText();`
     */
        timeout: 5000,
    },
    /* Run tests in files in parallel */
    fullyParallel: false,
    /* Fail the build on CI if you accidentally left test.only in the source code. */
    forbidOnly: !!process.env.CI,
    /* Retry on CI only */
    retries: process.env.CI ? 2 : 0,
    /* Opt out of parallel tests on CI. */
    workers: process.env.CI ? 1 : undefined,
    /* Reporter to use. See https://playwright.dev/docs/test-reporters */
    // reporter: [['json',
    //     {
    //         outputFile: 'report.json',
    //         open: 'never',
    //     },
    // ]],
    reporter: [
        ['html', {
            outputFile: 'report.html',
            open: 'never',
        }],
    ],
    // reporter: 'html',
    /* Shared settings for all the projects below. See https://playwright.dev/docs/api/class-testoptions. */
    use: {
    /* Maximum time each action such as `click()` can take. Defaults to 0 (no limit). */
        actionTimeout: 0,
        /* Base URL to use in actions like `await page.goto('/')`. */
        baseURL: process.env.EX_BASE_URI,

        /* Collect trace when retrying the failed test. See https://playwright.dev/docs/trace-viewer */
        trace: 'on-first-retry',
        headless: false,
        screenshot: 'on',
        video: 'on',

    },

    /* Configure projects for major browsers */
    projects: [
        {
            name: 'chromium',

            use: {
                ...devices['Desktop Chrome'],

            },
        },
    ],

    // outputDir: 'test-results/',
};

module.exports = config;

emailForwarding.spec.js

test.describe('Total cases', {
  test('First case', {
      console.log('Case1');
  });
 test('Second case', {
      console.log('Case2');
  });
});

webContent.spec.js

test.describe('Total cases', {
  test('First case', {
      console.log('Case1');
  });
 test('Second case', {
      console.log('Case2');
  });
});
